src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala

```diff
@@ -0,0 +1,283 @@
+   * has Succeeded and there are outputs, outputsOption will contain the response from the exec service.
+   */
+  case class ExecutionServiceStatusResponse(statusResponse: Seq[Try[Option[(WorkflowRecord, Option[ExecutionServiceOutputs])]]]) extends SubmissionMonitorMessage
+  case class StatusCheckComplete(terminateActor: Boolean) extends SubmissionMonitorMessage
+}
+
+/**
+ * An actor that monitors the status of a submission. Wakes up every submissionPollInterval and queries
+ * the execution service for status of workflows that we don't think are done yet. For any workflows
+ * that are successful, query again for outputs. Once all workflows are done mark the submission as done
+ * and terminate the actor.
+ *
+ * @param submissionId id of submission to monitor
+ * @param datasource
+ * @param submissionPollInterval time between polls of db for all workflow statuses within submission
+ */
+class SubmissionMonitorActor(val workspaceName: WorkspaceName,
+                             val submissionId: UUID,
+                             val datasource: SlickDataSource,
+                             val executionServiceDAO: ExecutionServiceDAO,
+                             val credential: Credential,
+                             val submissionPollInterval: FiniteDuration) extends Actor with SubmissionMonitor with LazyLogging {
+
+  import context._
+
+  scheduleNextMonitorPass
+
+  override def receive = {
+    case StartMonitorPass =>
+      logger.debug(s"polling workflows for submission $submissionId")
+      queryExecutionServiceForStatus() pipeTo self
+    case response: ExecutionServiceStatusResponse =>
+      logger.debug(s"handling execution service response for submission $submissionId")
+      handleStatusResponses(response) pipeTo self
+    case StatusCheckComplete(terminateActor) =>
+      logger.debug(s"done checking status for submission $submissionId, terminateActor = $terminateActor")
+      if (terminateActor) stop(self)
+      else scheduleNextMonitorPass
+
+    case Status.Failure(t) => throw t // an error happened in some future, let the supervisor handle it
+  }
+
+  def scheduleNextMonitorPass: Cancellable = {
+    system.scheduler.scheduleOnce(submissionPollInterval, self, StartMonitorPass)
+  }
+
+}
+
+trait SubmissionMonitor extends FutureSupport with LazyLogging {
+  val workspaceName: WorkspaceName
+  val submissionId: UUID
+  val datasource: SlickDataSource
+  val executionServiceDAO: ExecutionServiceDAO
+  val credential: Credential
+  val submissionPollInterval: Duration
+
+  import datasource.dataAccess.driver.api._
+
+  /**
+   * This function starts a monitoring pass
+   * @param executionContext
+   * @return
+   */
+  def queryExecutionServiceForStatus()(implicit executionContext: ExecutionContext): Future[ExecutionServiceStatusResponse] = {
+    datasource.inTransaction { dataAccess =>
+      dataAccess.workflowQuery.listWorkflowRecsForSubmissionAndStatuses(submissionId, WorkflowStatuses.runningStatuses: _*)
```

Comment:
Try doing the explain, but leaving out the `SUBMISSION_ID = 0x043AE84815084E74A7C027ECC3308B6C` clause - in other words, just filtering by status. Explain shows it's doing a full scan on that. I suspect with both clauses, it filters first to the submissionId, then does a scan over the remainder of records. I don't know how much another index would optimize this, but I know for sure the statuses are not indexed.